addons/account/models/account_reconcile_model.py

```diff
@@ -273,12 +273,15 @@ def _get_taxes_move_lines_dict(self, tax, base_line_dict):
         new_aml_dicts = []
         for tax_res in res['taxes']:
             tax = self.env['account.tax'].browse(tax_res['id'])
+            balance = tax_res['amount']
 
             new_aml_dicts.append({
                 'account_id': tax_res['account_id'] or base_line_dict['account_id'],
                 'name': tax_res['name'],
                 'partner_id': base_line_dict.get('partner_id'),
-                'balance': tax_res['amount'],
+                'balance': balance,
+                'debit': balance > 0 and balance or 0,
+                'credit': balance < 0 and -balance or 0,
```

Comment:
I don't understand why you need to add `debit`/`credit`... Is it changing something js-side?

Reply:
Because the widgets depends on balance, debit and credit. Since the write-off values are now returned by _apply_rules, we need to return them here too. See https://github.com/odoo/enterprise/pull/12995/files#diff-401b12fbdb45d436abeb4f3564da7731R1005

I agree we could refactor more in depth, but let's keep that for master, and wan's task :)